HeterogeneousCore/AlpakaInterface/interface/prefixScan.h

```diff
@@ -188,8 +188,22 @@ namespace cms::alpakatools {
         psum[i] = (j < size) ? co[j] : T(0);
       }
       alpaka::syncBlockThreads(acc);
-      blockPrefixScan(acc, psum, psum, blocksPerGrid, ws);
-
+      if constexpr (!requires_single_thread_per_block_v<TAcc>) {
+        if (blocksPerGrid <= warpSize * warpSize)
+          blockPrefixScan(acc, psum, psum, blocksPerGrid, ws);
+        else {
+          auto off = 0u;
+          while (off + warpSize * warpSize < blocksPerGrid) {
+            blockPrefixScan(acc, psum + off, warpSize * warpSize, ws);
+            off = off + warpSize * warpSize - 1;
+            // ^ this -1 is to keep the previous round total sum around
+            alpaka::syncBlockThreads(acc);
+          }
+          blockPrefixScan(acc, psum + off, psum + off, blocksPerGrid - off, ws);
+        }
+      } else {
+        blockPrefixScan(acc, psum, psum, blocksPerGrid, ws);
```

Comment:
And this simply
```suggestion
        blockPrefixScan(acc, psum, blocksPerGrid, ws);
```
?